pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -55,9 +80,117 @@ func (r *ReconcileStorageCluster) ensureStorageClasses(instance *ocsv1.StorageCl
 			}
 		}
 	}
+	return nil
+}
 
-	instance.Status.StorageClassesCreated = true
-
+// ensureExternalStorageClusterResources ensures that requested resources for the external cluster
+// being created
+func (r *ReconcileStorageCluster) ensureExternalStorageClusterResources(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	// check for the status boolean value accepted or not
```

Comment:
```suggestion
	// if external storage is not enabled, this function should not be called
	if !instance.Spec.ExternalStorage.Enable {
		return nil
	}
	// check for the status boolean value accepted or not
```

Even though this function is being put into the `else` part (where `Spec.ExternalStorage.Enable` is true) of the reconcile method, as an independent function, it should only be called in an external storage environment.
@jarrpa , what do you think?